lib/matplotlib/hatch.py

```diff
@@ -179,21 +179,7 @@ def __init__(self, hatch, density):
     ]
 
 
-def _validate_hatch_pattern(hatch):
-    valid_hatch_patterns = set(r'-+|/\xXoO.*')
-    if hatch is not None:
-        invalids = set(hatch).difference(valid_hatch_patterns)
-        if invalids:
-            valid = ''.join(sorted(valid_hatch_patterns))
-            invalids = ''.join(sorted(invalids))
-            _api.warn_deprecated(
-                '3.4',
-                removal='3.9',  # one release after custom hatches (#20690)
```

Comment:
The removal of the expiration of the deprecation needs to be notes in the API changes.